src/Development/Shake/Command.hs

```diff
@@ -417,15 +417,31 @@ commandExplicitIO params = removeOptionShell params $ \params -> removeOptionFSA
             Just v -> do
                 v <- canonicalizePath v `catchIO` const (pure v)
                 pure $ "Current directory: " ++ v ++ "\n"
-        liftIO $ errorIO $
-            "Development.Shake." ++ funcName ++ ", system command failed\n" ++
-            "Command line: " ++ optRealCommand ++ "\n" ++
-            (if optRealCommand /= optUserCommand then "Original command line: " ++ optUserCommand ++ "\n" else "") ++
-            cwd ++
-            "Exit code: " ++ show (case exit of ExitFailure i -> i; _ -> 0) ++ "\n" ++
-            if null captured then "Stderr not captured because WithStderr False was used\n"
-            else if null exceptionBuffer then intercalate " and " captured ++ " " ++ (if length captured == 1 then "was" else "were") ++ " empty"
-            else intercalate " and " captured ++ ":\n" ++ unlines (dropWhile null $ lines $ concat exceptionBuffer)
+        liftIO $ throwIO $ CommandIOException funcName optRealCommand optUserCommand
+                                              cwd exit captured exceptionBuffer
+
+data CommandIOException =
```

Comment:
If we're going to export this, it ends up being part of the API, so can it have some documentation comments added to it - saying it comes from commands

Reply:
I added the documentation now.